app/src/main/java/com/android/joinme/model/serie/SeriesRepositoryCached.kt

```
@@ -0,0 +1,176 @@
+  }
+
+  /**
+   * Applies client-side filtering to cached series when offline. Mimics the behavior of Firestore
+   * queries for each filter type.
+   */
+  private fun applySerieFilter(
+      serieFilter: SerieFilter,
+      series: List<Serie>,
+      userId: String
+  ): List<Serie> {
+    return when (serieFilter) {
+      SerieFilter.SERIES_FOR_OVERVIEW_SCREEN -> {
+        // Overview: series where user is a participant
+        series.filter { serie -> serie.participants.contains(userId) }
+      }
+      SerieFilter.SERIES_FOR_HISTORY_SCREEN -> {
+        // History: expired series where user is a participant, sorted by date descending
+        series
+            .filter { serie -> serie.participants.contains(userId) && serie.isExpired() }
+            .sortedByDescending { it.date.toDate().time }
+      }
+      SerieFilter.SERIES_FOR_SEARCH_SCREEN -> {
+        // Search: upcoming public series where user is NOT a participant or owner
+        series.filter { serie ->
+          serie.visibility == Visibility.PUBLIC &&
+              serie.isUpcoming() &&
+              !serie.participants.contains(userId) &&
+              serie.ownerId != userId
+        }
+      }
+      SerieFilter.SERIES_FOR_MAP_SCREEN -> {
+        // Map: upcoming or active series
+        series.filter { serie ->
+          serie.isUpcoming() || (serie.isActive() && serie.participants.contains(userId))
+        }
+      }
+    }
+  }
+
+  override suspend fun getSerie(serieId: String): Serie {
+    val cached = serieDao.getSerieById(serieId)?.toSerie()
+
+    if (!networkMonitor.isOnline()) {
+      return cached
+          ?: throw OfflineException(
+              "Cannot fetch serie while offline and no cached version available")
+    }
+
+    return try {
+      withTimeout(FIRESTORE_TIMEOUT_MS) {
+        firestoreRepo.getSerie(serieId).also { serie -> serieDao.insertSerie(serie.toEntity()) }
+      }
+    } catch (e: Exception) {
+      cached ?: throw e
```

Inconsistent error handling between getSerie and getAllSeries

Why This is Confusing:

  Scenario 1: Network error fetching all series (online)
  - Result: Silently returns cache, user unaware

  Scenario 2: Network error fetching single serie (online)
  - Result: Throws exception, crashes or shows error UI

  User Experience:
  - Overview screen: Works fine (uses cache)
  - Detail screen for same serie: ERROR!
  - User thinks: "This app is buggy"

---

We fixed the behavior of this getSerie(), you were absolutly right the implementation was not correct. We now have an implementation like getAllSeries().